src/pages/docs/messages/annotations.mdx

```diff
@@ -0,0 +1,418 @@
+| encoding | This is typically empty, as all messages received from Ably are automatically decoded client-side using this value. However, if the message encoding cannot be processed, this attribute contains the remaining transformations not applied to the `data` payload. |
+| timestamp | The timestamp of when the message was received by Ably, as milliseconds since the Unix epoch. |
+
+
+## Publishing annotations <a id="publish" />
+
+To publish an annotation for a message, use the `annotations.publish()` method on a channel, passing either a [message](/docs/messages) instance or the `serial` of the message to annotate. This method will publish an annotation message with an action of `annotation.create`.
+
+The `clientId` specified in the [client options](/docs/api/realtime-sdk#client-options) will be associated with the published annotation.
+
+<Aside data-type='note'>
+Some summarization methods require the client to be identified with a `clientId` in order to publish annotations. For more information, see [annotation summaries](#annotation-summaries).
+</Aside>
+
+When publishing an annotation, specify the [annotation type](#annotation-types) on the `type` field of the annotation object.
+
+You can also optionally specify a `name` for the annotation which is used by certain [summarization methods](#annotation-summaries) that aggregate by `name`.
+
+<Code>
+```javascript
+// Create an Ably Realtime client specifying the clientId that will
+// be associated with annotations published by this client
+const realtime = new Ably.Realtime({ key: '{{API_KEY}}', clientId: 'my-client-id' });
+
+// Create a channel in a namespace called `annotations`
+// which has message annotations enabled 
+const channel = realtime.channels.get('annotations:example');
+
+// Publish an annotation for a message that flags it as delivered
+await channel.annotations.publish(message, {
+  type: 'receipts:flag.v1',
+  name: 'delivered'
+});
+
+// You can also use a message's serial number
+await channel.annotations.publish(message.serial, {
+  type: 'receipts:flag.v1', 
+  name: 'delivered'
+});
+```
+
+```nodejs
+// Create an Ably Realtime client specifying the clientId that will
+// be associated with annotations published by this client
+const realtime = new Ably.Realtime({ key: '{{API_KEY}}', clientId: 'my-client-id' });
+
+// Create a channel in a namespace called `annotations`
+// which has message annotations enabled 
+const channel = realtime.channels.get('annotations:example');
+
+// Publish an annotation for a message that flags it as delivered
+await channel.annotations.publish(message, {
+  type: 'receipts:flag.v1',
+  name: 'delivered'
+});
+
+// You can also use a message's serial number
+await channel.annotations.publish(message.serial, {
+  type: 'receipts:flag.v1', 
+  name: 'delivered'
+});
+```
+</Code>
```

Comment:
this is character-for-character identical to the `javascript` code sample. I'm pretty sure you don't need both, I think we have a fallback waterfall defined so that it'll show the most relevant code sample that exists, and ofc that'd be javascript for nodejs? Mark can confirm

Reply:
That would be nice, @m-hulbert is there a way to do this?